tools/git-hooks/prepare-commit-msg

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This script prepares the commit messages to include
+# the issued ID at the beginning of the message.
+
+branch_name=$(git rev-parse --abbrev-ref HEAD)
+
+if [[ "$branch_name" =~ ^iox-#[0-9]+ ]]; then
+    branch_name="$(echo "$branch_name" | grep -Eo 'iox-#[0-9]+')"
+else
+    echo "Warning: Invalid branch name: ${branch_name}."
```

Comment:
Should this maybe be an `Error: The branch name doesn't follow the convention of "iox-#ISSUE_NUMBER"`?

Do we also want to enforce dashes instead of underscores in the branch name?

Reply:
I tend to agree that at least the `iox-#ISSUE_NUMBER` in the branch name should have a standardized format. At least what i can see in the forks is that everyone sticks to that, so it make sense to me to "enforce" that a bit.